torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2728,8 +2726,7 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::_reduce_scatter_base(
       outputTensor, // outputTensor
       rank_, // rank
       "_reduce_scatter_base", // colName
-      tensor.numel() * // inSize
-          this->getSize(),
+      inputTensor.numel(), // inSize
```

Comment:
wondering why we should change this, but not for gather on line 2470.

Reply:
The change here can help us remove the unnecessary calculation. But for allgahter function on line 2470, there's no such straightforward way to get the result.